lib/get-post.js

```diff
@@ -0,0 +1,39 @@
+import dayjs from "dayjs"
+import { paramCase } from "param-case"
+const { NotionToMarkdown } = require("notion-to-md")
+
+import notion from "lib/notion"
+
+const getPost = async (pageId) => {
+  const page = await notion.pages.retrieve({ page_id: pageId })
+
+  const meta = {
+    slug: `${paramCase(
+      page.properties["Name"][page.properties["Name"].type][0]?.plain_text,
+    )}--${page.id}`,
+    title: page.properties["Name"][page.properties["Name"].type][0]?.plain_text,
+    description:
+      page.properties["Description"][page.properties["Description"].type][0]
+        ?.plain_text || null,
+    date: dayjs(
+      page.properties["Date"][page.properties["Date"].type].start,
+    ).format("DD MMMM YYYY"),
+    cover: page?.cover?.[page.cover.type].url || null,
+    redirect:
+      page.properties["Redirect"][page.properties["Redirect"].type][0]
+        ?.plain_text || null,
+  }
```

Comment:
```suggestion
  const getVal = (name, prop = 'plain_text') => {
    const unknownNode = page.properties[name][page.properties[name].type]
    const knownNode = Array.isArray(node) ? node[0] : node
    return knownNode?.[prop]
  }
  
  const meta = {
    slug: `${paramCase(getVal("Name"))}--${page.id}`,
    title: getVal("Name"),
    description: getVal("Description"),
    date: dayjs(getVal("Date", "start")).format("DD MMMM YYYY"),
    cover: page?.cover?.[page.cover.type].url ?? null,
    redirect: getVal("Redirect") ?? null,
  }
```

fais toi des helpers autrement c'est illisible